### ISSUE TITLE
Improvement of Keras model training time 

### ISSUE THREAD

**Issue by marticobos** [ComputerVisionCentre/mcv-m5#2]
Hello,

While performing first batch of CNN training tests we found out that there was a huge delay between epochs. We suspected this delay was produced due to the prediction of the validation image set at the end of every epoch. By digging into the code, we realized the "validation_steps" parameter of method "fit_generator"  was not set when training the net (file code/model/model.py, line 44). According to Keras documentation, when this parameter is not set, the images of the validation set are analyzed one by one.

In order to reduce training time by analyzing validation images in batches (and take profit from parallel computing), we propose to make use of "batch_size_valid" configuration parameter. We also propose to use "batch_size_train" configuration parameter to define the batch size  when training the CNN. The following .zip file contains the code/model/model.py with the modification proposed:

[model.py.zip](https://github.com/ComputerVisionCentre/mcv-m5/files/1768683/model.py.zip)

With this modification and a minor tweak of the batch size hyper-parameters, we decreased the training time from 10 minutes to 3 minutes per epoch. 

Here is the code snippet for this modification:

```
hist = self.model.fit_generator(generator=train_gen,
	steps_per_epoch=self.cf.dataset.n_images_train//self.cf.batch_size_train,
	epochs=self.cf.n_epochs,
	verbose=1,
	callbacks=cb,
	validation_data=valid_gen,
	validation_steps = self.cf.dataset.n_images_valid//self.cf.batch_size_valid,
	class_weight=None,
	max_q_size=10,
	nb_worker=1,
	pickle_safe=False)
```

**Comment by lluisgomez** [comment]
Yes, well spotted!

I've made a commit 755b54741be30b90a27b50c430131e7f34f70c2e myself because I've seen there is another function call affected by the same problem

@marticobos next time you spot a bug, please consider creating a pull request to keep track of who has contributed to the main repository, ok?

thank you!

**Comment by marticobos** [comment]
Ok @lluisgomez, next time I will do a pull request when spotting a bug.

Best regards.
